lib/services/presets.go

```diff
@@ -79,6 +79,7 @@ func NewPresetEditorRole() types.Role {
 					types.NewRule(types.KindInstance, RO()),
 					types.NewRule(types.KindLoginRule, RW()),
 					types.NewRule(types.KindSAMLIdPServiceProvider, RW()),
+					types.NewRule(types.KindPlugin, append(ReadNoSecrets(), types.VerbCreate, types.VerbUpdate, types.VerbDelete)),
```

Comment:
My reasoning here is that since auth exchanges the initial credentials, and the plugin (which interacts with auth server directly in this case) renews them, there is no reason to let even `editor` manipulate those.

This may or may not be overkill, let me know.

Reply:
I'll let someone else chime in on RW vs RO-plus-select-edits (it seems used for `KindDevice` at least, so it wouldn't be new) and I'll just say that it feels very weird for `editor` to only have `ReadNoSecrets`.

Reply:
So `Plugin` stores OAuth credentials associated with us (the app) & the user. If/when this is extended to self-hosted Enterprise customers, they would be setting up their OAuth app anyway. In Cloud however, these would be credentials issued for a "Teleport Cloud" app managed by us, meaning if a user can get the credentials, they can impersonate this app with our branding, albeit only in their own Slack workspace / PagerDuty account / etc.. 🙂 That's the reason I started with conservative permissions.

Would it be viable to change this preset to RW later on as needed?

Reply:
If that's a concern then the permission system is not the correct place where to put this restriction, it should be made structural.

Nothing here stops me from adding all  permissions for the plugin type (or any type, I typically just give my user permissions for the `*` verb on the `*` resource).

Reply:
We can probably make this `RW()`, and remove the ability to read secrets via the gRPC layer then? The plugins themselves will skip gRPC and access the service directly.